star-rx72n-firmware/lib/rx_spi_comm/src/rx_spi_comm.c

```
@@ -672,11 +753,102 @@ static rx_err_t internal_decode_frame(const rx_spi_comm_handle_t* handle,
+    return k_dispatch_error;
+  }
+
+  /* Pre-condition 2: Frame must be valid (NASA Rule 5) */
+  if (frame == NULL || err == NULL) {
+    if (err != NULL) {
+      *err = k_rx_err_invalid_arg;
+    }
+    return k_dispatch_error;
+  }
+
+  const rx_frame_type_t type = (rx_frame_type_t)frame->header.type;
```

Clang (14.0.6)

[warning] variable 'type' is not initialized

(cppcoreguidelines-init-variables)